src/onboarding/sendEmail.js

```diff
@@ -1,36 +1,40 @@
 function sendEmail(name, email) {
-  // sending onboarding email
-  var htmlTemplate = HtmlService.createTemplateFromFile("emailTemplate");
-  htmlTemplate.name = name;
-  htmlContent = htmlTemplate.evaluate().getContent();
-  var subject = name + "'s " + "Coderbunker Onboarding";
-
-  var profileImageFileId = PropertiesService.getScriptProperties().getProperty("PROFILE_IMAGE_FILE_ID");
-  var profileImage = DriveApp.getFileById(profileImageFileId);
-  var addDriveImageId = PropertiesService.getScriptProperties().getProperty("ADD_DRIVE_IMAGE_ID");
-  var addDriveImage = DriveApp.getFileById(addDriveImageId);
-
-  GmailApp.sendEmail(
-    email, 
-    subject, 
-    "", 
-    {
-      from: "services@coderbunker.com",
-      name: "Coderbunker Services", 
-      cc: "bizdev@coderbunker.com",
-      htmlBody: htmlContent, 
-      inlineImages: {
-        imageKey: profileImage.getAs(MimeType.PNG),
-        addDrive: addDriveImage.getAs(MimeType.PNG),
-      },
-    }
-  );
-
-  // save the email sending result
   var resultSheet = SpreadsheetApp.getActiveSpreadsheet().getSheetByName("Emails");
   var userRowIndex = searchRow(name, resultSheet);
   var newUserRowRange = resultSheet.getRange(userRowIndex, 3); // get the "Email sent" cell
-  newUserRowRange.setValue(new Date());
+  
+  if (!newUserRowRange.getValue()) {  // if there is no sending record
+    // sending onboarding email
+    var htmlTemplate = HtmlService.createTemplateFromFile("emailTemplate");
+    htmlTemplate.name = name;
+    htmlContent = htmlTemplate.evaluate().getContent();
+    var subject = name + "'s " + "Coderbunker Onboarding";
+    
+    var profileImageFileId = PropertiesService.getScriptProperties().getProperty("PROFILE_IMAGE_FILE_ID");
```

Comment:
parameters sent to the function instead of retrieving from here? try to limit the I/O within your functions, pass explicit parameters instead to prevent side-effects or harder to understand logic

Reply:
Done. create a new function called `getProperty()` in utils.gs